shell/platform/windows/windows_lifecycle_manager.cc

```diff
@@ -13,8 +13,38 @@
 
 namespace flutter {
 
-WindowsLifecycleManager::WindowsLifecycleManager(FlutterWindowsEngine* engine)
-    : engine_(engine), process_close_(false) {}
+constexpr char kChannelName[] = "flutter/lifecycle";
+constexpr char kEnableMethod[] = "setLifecycleStateEnabled";
+constexpr char kEnableKey[] = "lifecycleStateEnabled";
+constexpr char kLifecycleError[] = "Lifecycle State Error";
+constexpr char kInvalidArgumentMessage[] = "Invalid argument";
+
+WindowsLifecycleManager::WindowsLifecycleManager(BinaryMessenger* messenger,
+                                                 FlutterWindowsEngine* engine)
+    : engine_(engine), process_close_(false) {
+  if (messenger) {
+    channel_ = std::make_unique<MethodChannel<rapidjson::Document>>(
+        messenger, kChannelName, &JsonMethodCodec::GetInstance());
+    channel_->SetMethodCallHandler(
+        [this](const MethodCall<rapidjson::Document>& call,
+               std::unique_ptr<MethodResult<rapidjson::Document>> result) {
+          const std::string& method = call.method_name();
+          if (method.compare(kEnableMethod) == 0) {
+            const rapidjson::Value& arguments = call.arguments()[0];
+            rapidjson::Value::ConstMemberIterator itr =
+                arguments.FindMember(kEnableKey);
+            if (itr == arguments.MemberEnd() || !itr->value.IsBool()) {
+              result->Error(kLifecycleError, kInvalidArgumentMessage);
+              return;
+            }
+            engine_->SetAppLifecycleStateEnabled(itr->value.GetBool());
+            result->Success();
+            return;
+          }
+          result->NotImplemented();
+        });
```

Comment:
Should we consolidate all of the `flutter/lifecycle` platform channel stuffs to a lifecycle plugin? Specifically:

1. Listening to `flutter/lifecycle`'s `setLifecycleStateEnabled`
2. Sending `flutter/lifecycle` application states

Reply:
What logic is part of the engine currently other than deferring to calls in the lifecycle manager that you are referring to?

Reply:
I'm referring to just the message serialization/deserialization plus the platform channel  sending/receiving here, not the business logic that decides when to send messages or how to handle received messages.

Right now, serializing + sending `flutter/lifecycle` messages is done by the `FlutterWindowsEngine`, and receiving + deserializing `flutter/lifecycle` messages is done by the `WindowsLifecycleManager`. It would be preferable if this was consolidated. I'd be fine with consolidating this in `WindowsLifecycleManager`. If for whatever reason this isn't feasible, I'd consider creating a lifecycle plugin for the `flutter/lifecycle` message serialization/deserialization + platform channel sending/receiving.